plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastoreClient.java

```diff
@@ -177,6 +181,33 @@ public Database getDatabase(String dbName)
         return client.getDatabase(dbName);
     }
 
+    @Override
+    public Map<String, List<String>> getAllTables()
+            throws TException
+    {
+        return parseTableMetadata(client.getTableMeta("*", "*", ImmutableList.of()));
```

Comment:
Cool, I didn't know they had this API. We might need to use `alternativeCall()` since old metastores might not support this.

Reply:
It was supported from Hive 2.0 Ref : https://issues.apache.org/jira/browse/HIVE-7575. Do we need to support versions prior to that ?

Reply:
Some of our PTs fail on that so I guess we need to.
But I was wondering whether to just call it once at the beginning and see if it works instead of doing alternative call.
The cache needs to be set upfront and I don't want to complicate things with some fallback mechanisms.